cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -234,6 +234,55 @@ class aggregate_reader_metadata {
 
   [[nodiscard]] auto get_num_row_groups() const { return num_row_groups; }
 
+  /**
+   * @brief Checks if a schema index from 0th source is mapped to the specified file index
+   *
+   * @return True if schema index is mapped
+   */
+  [[nodiscard]] inline bool is_schema_index_mapped(int schema_idx, int pfm_idx) const
+  {
+    // Check if schema_idx or pfm_idx is invalid
+    if (schema_idx < 0 or pfm_idx < 0 or pfm_idx >= static_cast<int>(per_file_metadata.size())) {
```

Comment:
should we throw here as well?

Reply:
Yeah I think we can throw here as false doesn't make sense.